### PR TITLE
fix : add None filter in the summation

### LINKED ISSUE #5705: 404 error on free orders + paid orders
**Describe the bug**

Admin/Sales page throws a 404 Error when both free orders and paid orders are there

**To Reproduce**
image (1)

1. Make some free and paid orders
2. Go to Admin/Sales
3. See error

**Expected behavior**

It should not throw a 404 error

**Stacktrace**

image

**Additional details (please complete the following information):**
 - OS: [e.g. MacOS, Ubuntu, CentOS] Ubuntu
 - Python Version [e.g. `3.5`, `3.6`] 3.5

**Additional context**

Working on it

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5705 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Filters out `None` Type objects (`Free Order Amounts`)

#### Changes proposed in this pull request:

- add None filter in the amount summation

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review

File: app/api/admin_sales/utils.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def status_summary(orders, status):
     dictionary
     """
     return {
-        'sales_total': sum([o.amount for o in orders if o.status == status]),
+        'sales_total': sum([o.amount for o in orders if o.status == status and o.amount is not None]),
```
Comment (iamareebjamal) [inline app/api/admin_sales/utils.py]: Amount should never be None, make it non nullable

Comment (uds5501) [reply app/api/admin_sales/utils.py]: @iamareebjamal Okay sir, working on it

File: app/models/order.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ class Order(SoftDeletionModel):
 
     id = db.Column(db.Integer, primary_key=True)
     identifier = db.Column(db.String, unique=True)
-    amount = db.Column(db.Float)
+    amount = db.Column(db.Float, nullable=False, default=0)
```
Comment (iamareebjamal) [inline app/models/order.py]: There's no migration file. This change is a no-op

Comment (uds5501) [reply app/models/order.py]: @iamareebjamal I've added the migration in 4de8cf81650926690484c1b10c3e02e0f145fe16

Comment (iamareebjamal) [reply app/models/order.py]: The migration is handwritten

Comment (uds5501) [reply app/models/order.py]: @iamareebjamal It indeed is. I added the `op.execute()` part because any existence of `null` values in the database (which were present  there because of creation of orders of Free Tickets) failed the database upgrade command

Comment (iamareebjamal) [reply app/models/order.py]: If you set the default value in AlterColumn command, it won't fail. Migration tools are not made without keeping this very common case in mind

Comment (uds5501) [reply app/models/order.py]: @iamareebjamal I am using this migration file now :

```python

def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.alter_column('orders', 'amount',
               existing_type=postgresql.DOUBLE_PRECISION(precision=53),
               server_default=0,
               nullable=False)
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.alter_column('orders', 'amount',
               existing_type=postgresql.DOUBLE_PRECISION(precision=53),
               nullable=True)
    # ### end Alembic commands ###

```
and the `upgrade` command throws this error:

```
sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) column "amount" contains null values
 [SQL: 'ALTER TABLE orders ALTER COLUMN amount SET NOT NULL']

```

Comment (iamareebjamal) [reply app/models/order.py]: `server_default='0'`

It takes a string

Comment (uds5501) [reply app/models/order.py]: @iamareebjamal Still throws the same error :sweat_smile:

Comment (iamareebjamal) [reply app/models/order.py]: What's the SQL generated now?

Comment (uds5501) [reply app/models/order.py]: @iamareebjamal  This is the trace : 

```
Traceback (most recent call last):
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
psycopg2.IntegrityError: column "amount" contains null values


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 97, in <module>
    manager.run()
  File "/home/uddeshya/venv/lib/python3.5/site-packages/flask_script/__init__.py", line 417, in run
    result = self.handle(argv[0], argv[1:])
  File "/home/uddeshya/venv/lib/python3.5/site-packages/flask_script/__init__.py", line 386, in handle
    res = handle(*args, **config)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/flask_migrate/__init__.py", line 259, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/command.py", line 254, in upgrade
    script.run_env()
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/script/base.py", line 427, in run_env
    util.load_python_file(self.dir, 'env.py')
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/util/pyfiles.py", line 81, in load_python_file
    module = load_module_py(module_id, path)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/util/compat.py", line 83, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "migrations/env.py", line 158, in <module>
    run_migrations_online()
  File "migrations/env.py", line 151, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/runtime/environment.py", line 836, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/runtime/migration.py", line 330, in run_migrations
    step.migration_fn(**kw)
  File "/home/uddeshya/Desktop/mySetup/open-event-server/migrations/versions/0dca1ba3ad34_.py", line 24, in upgrade
    nullable=False)
  File "<string>", line 8, in alter_column
  File "<string>", line 3, in alter_column
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/operations/ops.py", line 1444, in alter_column
    return operations.invoke(alt)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/operations/base.py", line 319, in invoke
    return fn(self, operation)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/operations/toimpl.py", line 53, in alter_column
    **operation.kw
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/ddl/postgresql.py", line 118, in alter_column
    **kw)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/ddl/impl.py", line 145, in alter_column
    existing_nullable=existing_nullable,
  File "/home/uddeshya/venv/lib/python3.5/site-packages/alembic/ddl/impl.py", line 118, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 945, in execute
    return meth(self, multiparams, params)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1002, in _execute_ddl
    compiled
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
    context)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
    exc_info
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
    raise value.with_traceback(tb)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/home/uddeshya/venv/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) column "amount" contains null values
 [SQL: 'ALTER TABLE orders ALTER COLUMN amount SET NOT NULL']

```

Comment (iamareebjamal) [reply app/models/order.py]: I'll see

**Comment by uds5501** [comment]
@iamareebjamal Please review

File: app/api/admin_sales/utils.py (hunk 1 above)
Comment (iamareebjamal) [inline app/api/admin_sales/utils.py]: Revert this change now

Comment (uds5501) [reply app/api/admin_sales/utils.py]: @iamareebjamal Done

**Comment by uds5501** [comment]
@CosmicCoder96 Please review :)
